tests/models/pop2piano/test_modeling_pop2piano.py

```diff
@@ -0,0 +1,668 @@
+
+        attention_mask = ids_tensor(
+            [self.model_tester.batch_size, self.model_tester.encoder_seq_length, self.model_tester.encoder_seq_length],
+            vocab_size=2,
+        )
+        decoder_attention_mask = ids_tensor(
+            [self.model_tester.batch_size, self.model_tester.decoder_seq_length, self.model_tester.decoder_seq_length],
+            vocab_size=2,
+        )
+
+        self.model_tester.create_and_check_decoder_model_attention_mask_past(
+            config,
+            input_ids,
+            decoder_input_ids,
+            attention_mask,
+            decoder_attention_mask,
+            lm_labels,
+        )
+
+    def test_decoder_model_past_with_large_inputs(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_decoder_model_past_large_inputs(*config_and_inputs)
+
+    def test_encoder_decoder_shared_weights(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_encoder_decoder_shared_weights(*config_and_inputs)
+
+    @unittest.skipIf(torch_device == "cpu", "Cant do half precision")
+    def test_model_fp16_forward(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_model_fp16_forward(*config_and_inputs)
+
+    def test_v1_1_resize_embeddings(self):
+        config = self.model_tester.prepare_config_and_inputs()[0]
+        self.model_tester.check_resize_embeddings_Pop2Piano_v1_1(config)
+
+    @slow
+    def test_model_from_pretrained(self):
+        for model_name in POP2PIANO_PRETRAINED_MODEL_ARCHIVE_LIST[:1]:
+            model = Pop2PianoForConditionalGeneration.from_pretrained(model_name)
+            self.assertIsNotNone(model)
+
+    @unittest.skip("Test has a segmentation fault on torch 1.8.0")
+    def test_export_to_onnx(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        model = Pop2PianoForConditionalGeneration(config_and_inputs[0]).to(torch_device)
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            torch.onnx.export(
+                model,
+                (config_and_inputs[1], config_and_inputs[3], config_and_inputs[2]),
+                f"{tmpdirname}/Pop2Piano_test.onnx",
+                export_params=True,
+                opset_version=9,
+                input_names=["input_ids", "decoder_input_ids"],
+            )
+
+    @unittest.skip("Does not work on the tiny model as we keep hitting edge cases.")
+    def test_disk_offload(self):
+        pass
+
+    @unittest.skip("Pop2Piano does not need this test.")
+    def test_generate_with_head_masking(self):
+        pass
+
+    @unittest.skip("Pop2Piano does not need this test.")
+    def test_generate_with_past_key_values(self):
+        pass
+
+
+@require_torch
+class Pop2PianoModelIntegrationTests(unittest.TestCase):
```

Comment:
We need more tests related to actual music files!

Reply:
@susnato 
Since most training datasets consist of commercial pop music, I don't think it's appropriate to add training dataset sample to a commercial repository like HF. 

Btw what is the purpose of this test script? If short (<5~10s) music is enough for test, then I can add a sample commercial music.

Reply:
@sweetcocoa  I think we should add the sample music to original HF pop2piano repo,to make sure all the files are in the same place. Please send me the link or the music file through mail I will add it to `susnato/pop2piano_dev`, later after the PR is finalized we can transfer all the files to original repo(`sweetcocoa/pop2piano`)